external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java

```diff
@@ -170,10 +170,10 @@ public void run() {
         }
     }
 
-    private void flushAllWriters()
+    private void flushAllWriters(boolean rollToNext)
         throws HiveWriter.CommitFailure, HiveWriter.TxnBatchFailure, HiveWriter.TxnFailure, InterruptedException {
         for(HiveWriter writer: allWriters.values()) {
-            writer.flush(true);
+            writer.flush(rollToNext);
```

Comment:
@caofangkun closing Writer handled in different method you can flushAllAndCloseWriters().